mysql-test/suite/rocksdb/t/use_direct_reads_writes.test

```diff
@@ -3,45 +3,29 @@
 # Issue221
 # Turning on both --rocksdb-allow-mmap-reads and --rocksdb-use-direct-reads
 # caused an assertion in RocksDB.  Now it should not be allowed and the
-# server will not start with that configuration
+# plugin will not initialize with that configuration
 
-# Write file to make mysql-test-run.pl expect the "crash", but don't restart
-# the server until it is told to
---let $_server_id= `SELECT @@server_id`
---let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.$_server_id.expect
---exec echo "wait" >$_expect_file_name
-shutdown_server 10;
+--let SEARCH_FILE=$MYSQLTEST_VARDIR/tmp/rocksdb.use_direct_reads_writes.err
 
-# Clear the log
---exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
-
-# Attempt to restart the server with invalid options
---exec echo "restart:--rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1" >$_expect_file_name
---sleep 0.1  # Wait 100ms - that is how long the sleep is in check_expected_crash_and_restart
---exec echo "restart:" >$_expect_file_name
-
-# Cleanup
---enable_reconnect
---source include/wait_until_connected_again.inc
---disable_reconnect
+# Expect MyRocks to fail to initialize with invalid combination of arguments
+--let $restart_parameters=restart:--log-error=$SEARCH_FILE --rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1
+--replace_result $MYSQLTEST_VARDIR MYSQLTEST_VARDIR
+--source include/restart_mysqld.inc
 
 # We should now have an error message
---exec grep "enable both use_direct_reads" $MYSQLTEST_VARDIR/log/mysqld.1.err | cut -d] -f2
+--let SEARCH_PATTERN=enable both use_direct_reads
+--source include/search_pattern_in_file.inc
 
 # Repeat with --rocksdb-use-direct-writes
---let $_server_id= `SELECT @@server_id`
---let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.$_server_id.expect
---exec echo "wait" >$_expect_file_name
-shutdown_server 10;
-
---exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
+--let $restart_parameters=restart:--log-error=$SEARCH_FILE --rocksdb_use_direct_io_for_flush_and_compaction=1 --rocksdb_allow_mmap_writes=1
+--replace_result $MYSQLTEST_VARDIR MYSQLTEST_VARDIR
+--source include/restart_mysqld.inc
 
---exec echo "restart:--rocksdb_use_direct_writes=1 --rocksdb_allow_mmap_writes=1" >$_expect_file_name
---sleep 0.1
---exec echo "restart:" >$_expect_file_name
-
---enable_reconnect
---source include/wait_until_connected_again.inc
---disable_reconnect
+# We should now have an error message
+--let SEARCH_PATTERN=enable both use_direct_io_for_flush_and_compaction
```

Comment:
Is there a chance of SEARCH_PATTERN mix-up, i.e. 2nd restart picking up something printed by the 1st one? If none at all, OK to keep as-is, otherwise the 2nd restart should use a different error log